controllers/controllers/services/instances/managed/controller_test.go

```diff
@@ -499,16 +499,75 @@ var _ = Describe("CFServiceInstance", func() {
 	})
 
 	When("the instance is deleted", func() {
+		BeforeEach(func() {
+			brokerClient.DeprovisionReturns(osbapi.ServiceInstanceOperationResponse{
+				Operation: "deprovision-op",
+			}, nil)
+			brokerClient.GetServiceInstanceLastOperationReturns(osbapi.LastOperationResponse{
+				State: "in progress",
+			}, nil)
+		})
+
 		JustBeforeEach(func() {
-			Expect(adminClient.Delete(ctx, instance)).To(Succeed())
+			// For deletion test we want to request deletion and verify the behaviour when finalization fails.
+			// Therefore we use the standard k8s client instncea of `adminClient` as it ensures that the object is deleted
+			Expect(k8sManager.GetClient().Delete(ctx, instance)).To(Succeed())
 		})
 
-		It("deletes the instance", func() {
+		It("deprovisions the instance with the broker", func() {
 			Eventually(func(g Gomega) {
 				err := adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)
 				g.Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+
+				g.Expect(brokerClient.DeprovisionCallCount()).To(Equal(1))
+				_, actualDeprovisionRequest := brokerClient.DeprovisionArgsForCall(0)
+				Expect(actualDeprovisionRequest).To(Equal(osbapi.InstanceDeprovisionPayload{
+					ID: instance.Name,
+					InstanceDeprovisionRequest: osbapi.InstanceDeprovisionRequest{
+						ServiceId: "service-offering-id",
+						PlanID:    "service-plan-id",
+					},
+				}))
+			}).Should(Succeed())
+		})
+
+		It("does not delete the service instance", func() {
+			Consistently(func(g Gomega) {
+				g.Expect(adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)).To(Succeed())
 			}).Should(Succeed())
 		})
+
+		When("deprovision fails", func() {
+			BeforeEach(func() {
+				brokerClient.DeprovisionReturns(osbapi.ServiceInstanceOperationResponse{}, errors.New("deprovision-failed"))
+			})
+
+			It("sets ready condition to false", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)).To(Succeed())
+					g.Expect(instance.Status.Conditions).To(ContainElement(SatisfyAll(
+						HasType(Equal(korifiv1alpha1.StatusConditionReady)),
+						HasStatus(Equal(metav1.ConditionFalse)),
+						HasReason(Equal("DeprovisionFailed")),
+					)))
+				}).Should(Succeed())
+			})
+		})
+
+		When("deprovision operation completes", func() {
```

Comment:
As the finalizer does not check last operation, this `When` does not seem to make sense